WCSimFLOWER.cpp

```diff
@@ -285,35 +331,33 @@ void WCSimFLOWER::GetNearestNeighbours(bool overwrite_root_file)
     WCSimRootPMT pmt, otherPMT;
     int tubeID_j;
     float x, y, z;
+    // Loop over 50cm PMTs to find their neighbours (don't search for neighbours of mPMTs, see https://github.com/HKDAQ/FLOWER/issues/12)
     for (unsigned int ipmt = 0; ipmt < fNPMTs; ipmt++) {
       if(fVerbose > 0 && ipmt % (fNPMTs / 10) == 0)
-	cout << "Finding nearest neighbours for PMT " << ipmt << " of " << fNPMTs << endl;
+      	cout << "Finding nearest neighbours for PMT " << ipmt << " of " << fNPMTs << endl;
       pmt = fGeom->GetPMT(ipmt);
       tubeID = pmt.GetTubeNo();
       x = pmt.GetPosition(0);
       y = pmt.GetPosition(1);
       z = pmt.GetPosition(2);
       if(fVerbose > 3)
-	cout << "Tube " << tubeID << " x,y,z " << x << "," << y << "," << z << endl;
+	      cout << "Tube " << tubeID << " x,y,z " << x << "," << y << "," << z << endl;
 
-      // loop over all PMTs and get the IDs of each ones that are "close"
-      // In a 3x3 grid around PMT 'tubeID'
-      // explitly uses distance to neighboring PMTs 
-      //  (for 40% coverage with 50cm PMTs, this is 70.71 cm (100 cm diagonally))
+      // loop over all PMTs and get the IDs of each ones that are closer that fNeighbourDistance
       neighbours.clear();
-      for (unsigned int jpmt = 0; jpmt < fNPMTs; jpmt++) {
-	if(jpmt == ipmt) continue; // don't count the current PMT itself
-	otherPMT = fGeom->GetPMT(jpmt);
-	tubeID_j = otherPMT.GetTubeNo();
-	if (sqrt(pow(x - otherPMT.GetPosition(0), 2) + 
-		 pow(y - otherPMT.GetPosition(1), 2) +
-		 pow(z - otherPMT.GetPosition(2), 2)) < fNeighbourDistance) {
-	  neighbours.push_back(tubeID_j);
-	}
+      for (unsigned int jpmt = 0; jpmt < fNallPMTs; jpmt++) {
+        if(jpmt == ipmt) continue; // don't count the current PMT itself
+        otherPMT = fGeom->GetPMT(jpmt);
```

Comment:
This will not work for the 3inch PMTs unfortunately (except if you want to use the Geometry object from hk-AstroAnalysis).
In my opinion the easiest modification would be:
if ( jmpt < FIRST_PMT2 ) otherPMT = fGeom->GetPMT(jpmt);
else otherPMT = fGeom->GetPMT(jpmt - FIRST_PMT2, true);

With FIRST_PMT2 being 100000 currently.

Reply:
(maybe it should be <= instead of <)